cmd/helm/repo_list.go

```diff
@@ -39,7 +39,7 @@ func newRepoListCmd(out io.Writer) *cobra.Command {
 		ValidArgsFunction: noCompletions,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			f, err := repo.LoadFile(settings.RepositoryConfig)
-			if isNotExist(err) || (len(f.Repositories) == 0 && !(outfmt == output.JSON || outfmt == output.YAML)) {
+			if (isNotExist(err) || len(f.Repositories) == 0) && !(outfmt == output.JSON || outfmt == output.YAML) {
```

Comment:
I think this code will have the desired effect, but I also think it is a bit complicated to understand.

The entire `err` variable is more or less redundant here, as `len(f.Repositories) == 0` will always also be true if `isNotExist(err)` is true, so this could be simplified by removing it from the condition. 

Probably comes down to preference, but I find the solution in https://github.com/helm/helm/pull/10519 easier to understand.

Reply:
That's great. There are two solutions for one issue.The difference between the solution is not very big.
This PR keep the error, less code change. And https://github.com/helm/helm/pull/10519 is simpler.

Reply:
For what it's worth: I'm with Sönke and I like the other solution better.

Reply:
I'm not sure I completely follow your comment, but I'd personally be fine merging either of these PRs, I think they'll both solve the issue :)

Reply:
Should we merge it.

Reply:
fine for me